components/dashboard/src/workspaces/Workspaces.tsx

```diff
@@ -52,38 +44,18 @@ export default function () {
     const showStartWSModal = () => setIsTemplateModelOpen(true);
     const hideStartWSModal = () => setIsTemplateModelOpen(false);
 
-    const getRecentSuggestions: () => WsStartEntry[] = () => {
-        if (workspaceModel) {
-            const all = workspaceModel.getAllFetchedWorkspaces();
-            if (all && all.size > 0) {
-                const index = new Map<string, WsStartEntry & { lastUse: string }>();
-                for (const ws of Array.from(all.values())) {
-                    const repoUrl = Workspace.getFullRepositoryUrl(ws.workspace);
-                    if (repoUrl) {
-                        const lastUse = WorkspaceInfo.lastActiveISODate(ws);
-                        let entry = index.get(repoUrl);
-                        if (!entry) {
-                            entry = {
-                                title: Workspace.getFullRepositoryName(ws.workspace) || repoUrl,
-                                description: repoUrl,
-                                startUrl: gitpodHostUrl.withContext(repoUrl).toString(),
-                                lastUse,
-                            };
-                            index.set(repoUrl, entry);
-                        } else {
-                            if (entry.lastUse.localeCompare(lastUse) < 0) {
-                                entry.lastUse = lastUse;
-                            }
-                        }
-                    }
-                }
-                const list = Array.from(index.values());
-                list.sort((a, b) => b.lastUse.localeCompare(a.lastUse));
-                return list;
+    useEffect(() => {
+        const onKeyDown = (event: KeyboardEvent) => {
```

Comment:
**question**: Noticed we close the modal with `ESC` but what do you think of also closing the modal with the same key binding?

Reply:
Interesting. Is this a standard/expected behavior for repeat keystrokes?

I notice that VS Code does not close the command palette if you press `F1` a second time (or `Cmd+P` a second time).

Reply:
Again, like in https://github.com/gitpod-io/gitpod/pull/7715#discussion_r790696114:

> I've seen both approaches work equally good depending on the application.

> However, let's leave this as is for now and come back to this if we run into this in action. 🤝